modules/openapi/src/internals/IModelToSmithy.scala

```diff
@@ -46,11 +48,26 @@ final class IModelToSmithy(useEnumTraitSyntax: Boolean)
     val shapes = iModel.definitions.map[Shape] {
       case Structure(id, fields, _, structHints) =>
         val members = fields.map { case Field(id, tpe, hints) =>
+          val memName = id.memberName.value.toString
+          val nameWillNeedChange =
```

Comment:
minor, but if `nameWillNeedChange` is false all that header/query logic can be ignored

most probably overkill but if you do `def isHeader = ...` and `def isQuery = ...` and replace the order of `!isHeader && !isQuery && nameWillNeedChange` for `nameWillNeedChange && !isHeader && !isQuery`
you'd skip the evaluation of `isQuery` and `isHeader`

and as I was writing this, I realized that you can probably combine `isHeader` and `isQuery` to go through the hints list only once, because we're looking for one or the other

at the end of the day though, it may just obfuscate the code more

Reply:
Those are good points, I think it would still be plenty readable that way. I will switch it around